File: app/models/map.rb
class Map
  def get_welcome_markers(profiles)
    @arr = []
    profiles.each do |profile|
      if profile.user.name.nil?
        name = profile.user.email.gsub(/@+\w+.+\z/, '')
      else
        name = profile.user.name
      end
      @arr << { user_id: profile.user_id, name: name, summary: profile.summary, latitude: profile.latitude,
        longitude: profile.longitude, framework: profile.framework }
    end

    @arr
  end

  def get_user_marker(user)
    profile = user.profile
    @arr = []
    if profile.user.name.nil?
      name = profile.user.email.gsub(/@+\w+.+\z/, '')
    else
      name = profile.user.name
    end
    @arr << { name: name, summary: profile.summary, latitude: profile.latitude,
      longitude: profile.longitude, framework: profile.framework }

    @arr
  end
end

File: app/assets/javascripts/map.js
$('#map').ready(function () {
  // Load gmaps with search results
  $('#search-button').on('click', function() {
    setTimeout(function(){
      updatedResults = $('#results').data('results');
      console.log(updatedResults);
      initMap(); // Initialize map using updatedResults
    }, 1000);
  });

  function initMap() {
    var coord, map;
    coord = {
      lat: 42.3725019,
      lng: -71.1159379
      };
    map = new google.maps.Map(document.getElementById('map'), {
      zoom: 8,
      center: coord
    });
    jQuery.each(updatedResults, function() {
      var contentString, customIcon, infoWindow, marker, userCoord;
      userCoord = {
        lat: this.latitude,
        lng: this.longitude
      };
      contentString = '<div id="content">' + '<div id="siteNotice">' + '</div>' + '<h1 id="firstHeading" class="firstHeading">' + '<a href="http://localhost:3000/users/' + this.user_id + '">' + this.name + '</a>' + '</h1>' + '<div id="bodyContent">' + this.summary + '</div>' + '</div>';
      infoWindow = new google.maps.InfoWindow({
        content: contentString
      });
      if ((this.framework === null) || (this.framework === "Unspecified")) {
        customIcon = 'assets/jobs.png';
      } else {
        customIcon = 'assets/' + this.framework + '.png';
      }
      marker = new google.maps.Marker({
        position: userCoord,
        map: map,
        icon: customIcon,
        animation: google.maps.Animation.DROP,
        title: this.name
      });
      marker.addListener('click', function() {
        infoWindow.open(map, marker);
      });
    });
    return
  }
});
